tests/unit/parsers/test.manifestjson.js

```diff
@@ -3185,4 +3185,276 @@ describe('ManifestJSONParser', () => {
       );
     });
   });
+
+  describe('restricted permissions', () => {
+    const validate = ({ manifestProps, restrictedPermissions }) => {
+      const linter = new Linter({ _: ['bar'] });
+      const parser = new ManifestJSONParser(
+        validManifestJSON(manifestProps),
+        linter.collector,
+        { restrictedPermissions }
+      );
+
+      return { parser, linter };
+    };
+
+    it.each([
+      [
+        'no browser_specific_settings',
+        // manifest props
+        {
+          permissions: ['alarms'],
+        },
+        // restricted permissions
+        {
+          alarms: '88.1.3',
+        },
+      ],
+      [
+        'no browser_specific_settings.gecko',
+        // manifest props
+        {
+          browser_specific_settings: {},
+          permissions: ['alarms'],
+        },
+        // restricted permissions
+        {
+          alarms: '88.1.3',
+        },
+      ],
+      [
+        'no strict_min_version',
+        // manifest props
+        {
+          browser_specific_settings: {
+            gecko: {},
+          },
+          permissions: ['alarms'],
+        },
+        // restricted permissions
+        {
+          alarms: '88.1.3',
+        },
+      ],
+      [
+        'invalid strict_min_version (version below the strict min)',
+        // manifest props
+        {
+          browser_specific_settings: {
+            gecko: {
+              strict_min_version: '88.0',
+            },
+          },
+          permissions: ['alarms'],
+        },
+        // restricted permissions
+        {
+          alarms: '88.1.3',
+        },
+      ],
+      [
+        'invalid strict_min_version (version below the strict min)',
+        // manifest props
+        {
+          browser_specific_settings: {
+            gecko: {
+              strict_min_version: '88.1.2',
```

Comment:
> include in the test case at least one with a non numeric part, because it seems to be technically supported

Yeah, why not. It should be covered by addons-moz-compare already but adding a test case or two is easy.

Reply:
> double-check the scenario where the strict_min_version has an invalid format, to be sure that when we got to validate the permissions our assumption would not make us to break instead of behaving as expected (I think if the strict_min_version is invalid we still want to collect an additional RESTRICTED_PERMISSION lint error)

I don't know about collecting the error for the restricted permission. Maybe? There will definitely be an error reported for the invalid strict min version. I will add new tests anyway.

Reply:
The added test cases show that we collect two errors when strict_min_version is invalid.